sale_delivery_split_date/models/stock_picking.py

```diff
@@ -3,16 +3,38 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 from openerp import api, fields, models
+from openerp.osv import fields as old_fields
 
 
 class StockPicking(models.Model):
 
     _inherit = 'stock.picking'
 
+    def _get_sale_id(self, cr, uid, ids, name, args, context=None):
+        res = {}
+        line_obj = self.pool.get('sale.order.line')
+        for picking in self.browse(cr, uid, ids, context=context):
+            res[picking.id] = super(StockPicking, self)._get_sale_id(
+                cr, uid, picking.ids, name, args, context=context)[picking.id]
+            if not res[picking.id] and picking.group_id:
+                line_id = line_obj.search(
+                    cr, uid,
+                    [('procurement_group_id', '=', picking.group_id.id)],
+                    limit=1, context=context)
+                line = line_obj.browse(cr, uid, line_id, context=context)
+                res[picking.id] = line.order_id
+        return res
+
     min_dt = fields.Date(
         string='Scheduled Date (for filter purpose only)',
         compute='_compute_min_dt', store=True)
 
+    _columns = {
+        'sale_id': old_fields.function(
```

Comment:
Why do you need to add this field? This field already exists.

Reply:
If I'm not mistaken when the field is a function (old API) you need to define again in order to extend the function itself. Not sure, and I haven't been able to prove.

Reply:
Ah, ok, I am not proficient with old api